service/frontend/workflowHandler.go

```diff
@@ -3216,23 +3209,39 @@ func (wh *WorkflowHandler) getArchivedHistory(
 	if err != nil {
 		return nil, wh.error(err, scope)
 	}
-	archivalBucket := entry.GetConfig().HistoryArchivalURI // TODO ycyang: rewrite get archived history with archiver
-	if archivalBucket == "" {
+
+	archivalURI := entry.GetConfig().HistoryArchivalURI
+	if archivalURI == "" {
 		return nil, wh.error(errHistoryHasPassedRetentionPeriod, scope)
 	}
-	downloadReq := &warchiver.DownloadBlobRequest{
-		NextPageToken:  request.NextPageToken,
-		ArchivalBucket: archivalBucket,
-		DomainID:       domainID,
-		WorkflowID:     request.GetExecution().GetWorkflowId(),
-		RunID:          request.GetExecution().GetRunId(),
+
+	scheme, err := common.GetArchivalScheme(archivalURI)
+	if err != nil {
+		return nil, wh.error(err, scope)
 	}
-	resp, err := wh.historyBlobDownloader.DownloadBlob(ctx, downloadReq)
+
+	historyArchiver, err := wh.archiverProvider.GetHistoryArchiver(scheme, common.FrontendServiceName)
 	if err != nil {
 		return nil, wh.error(err, scope)
 	}
+
+	resp, err := historyArchiver.Get(ctx, archivalURI, &archiver.GetHistoryRequest{
+		DomainID:      domainID,
+		WorkflowID:    request.GetExecution().GetWorkflowId(),
+		RunID:         request.GetExecution().GetRunId(),
+		NextPageToken: request.GetNextPageToken(),
+		PageSize:      int(request.GetMaximumPageSize()),
```

Comment:
We are treating pageSize as number of events is this how persistence treats it? I thought in persistence it represents number of batches?

Reply:
I think it's also number of events in persistence.